htdocs/content/shipyard.php

```diff
@@ -654,44 +633,43 @@
 
+                        //Rohstoffe summieren, diese werden nach der Schleife abgezogen
+                        $totalMetal += $bc['metal'];
+                        $totalCrystal += $bc['crystal'];
+                        $totalPlastic += $bc['plastic'];
+                        $totalFuel += $bc['fuel'];
+                        $totalFood += $bc['food'];
+
                         //Log schreiben
                         $log_text = "[b]Schiffsauftrag Bauen[/b]
 
-                        [b]Start:[/b] ".date("d.m.Y H:i:s",$end_time)."
-                        [b]Ende:[/b] ".date("d.m.Y H:i:s",$end_time)."
-                        [b]Dauer:[/b] ".tf($duration)."
-                        [b]Dauer pro Einheit:[/b] ".tf($obj_time)."
-                        [b]Schiffswerft Level:[/b] ".CURRENT_SHIPYARD_LEVEL."
-                        [b]Eingesetzte Bewohner:[/b] ".nf($people_working)."
-                        [b]Gen-Tech Level:[/b] ".$gen_tech_level."
-                        [b]Eingesetzter Spezialist:[/b] ".$cu->specialist->name."
+                        [b]Start:[/b] " . date("d.m.Y H:i:s", $end_time) . "
+                        [b]Ende:[/b] " . date("d.m.Y H:i:s", $end_time) . "
+                        [b]Dauer:[/b] " . tf($duration) . "
+                        [b]Dauer pro Einheit:[/b] " . tf($obj_time) . "
+                        [b]Schiffswerft Level:[/b] " . CURRENT_SHIPYARD_LEVEL . "
+                        [b]Eingesetzte Bewohner:[/b] " . nf($people_working) . "
+                        [b]Gen-Tech Level:[/b] " . $gen_tech_level . "
+                        [b]Eingesetzter Spezialist:[/b] " . $cu->specialist->name . "
 
                         [b]Kosten[/b]
-                        [b]".RES_METAL.":[/b] ".nf($bc['metal'])."
-                        [b]".RES_CRYSTAL.":[/b] ".nf($bc['crystal'])."
-                        [b]".RES_PLASTIC.":[/b] ".nf($bc['plastic'])."
-                        [b]".RES_FUEL.":[/b] ".nf($bc['fuel'])."
-                        [b]".RES_FOOD.":[/b] ".nf($bc['food'])."
+                        [b]" . RES_METAL . ":[/b] " . nf($bc['metal']) . "
+                        [b]" . RES_CRYSTAL . ":[/b] " . nf($bc['crystal']) . "
+                        [b]" . RES_PLASTIC . ":[/b] " . nf($bc['plastic']) . "
+                        [b]" . RES_FUEL . ":[/b] " . nf($bc['fuel']) . "
+                        [b]" . RES_FOOD . ":[/b] " . nf($bc['food']) . "
 
                         [b]Rohstoffe auf dem Planeten[/b]
-                        [b]".RES_METAL.":[/b] ".nf($planet->resMetal)."
-                        [b]".RES_CRYSTAL.":[/b] ".nf($planet->resCrystal)."
-                        [b]".RES_PLASTIC.":[/b] ".nf($planet->resPlastic)."
-                        [b]".RES_FUEL.":[/b] ".nf($planet->resFuel)."
-                        [b]".RES_FOOD.":[/b] ".nf($planet->resFood)."";
+                        [b]" . RES_METAL . ":[/b] " . nf($planet->resMetal - $totalMetal) . "
```

Comment:
Could also fetch the planet after we change any data to make sure it is up to date

Reply:
As for later, ideally every such request would be POST/PUT/DELET and then redirect to e.g. the shipyard. This way the page that renders will always have data that is up to date.